setup.py

```diff
@@ -111,6 +111,7 @@
         "PyYAML>=5.0,<7",
         "requests>=2.28.1,<3",
         "rich>=12.5.1,<14",
+        "setuptools",  # Installation tool
```

Comment:
are we using setuptools? or just means as a downstream dependency of trying to install a plugin?

Reply:
tbh can probably add a TODO to remove when migrating to pyproject.toml workflow for all plugins

Reply:
without it `ape plugins list` crashes, which seems pretty key (due to pkg_resources)

```
>> ape plugins list 
Traceback (most recent call last):
  File "/code/newape/.venv/bin/ape", line 5, in <module>
    from ape._cli import cli
  File "/code/newape/.venv/lib/python3.10/site-packages/ape/__init__.py", line 10, in <module>
    from ape.managers.project import ProjectManager as Project
  File "/code/newape/.venv/lib/python3.10/site-packages/ape/managers/__init__.py", line 23, in <module>
    ManagerAccessMixin.config_manager = ConfigManager(
  File "/code/newape/.venv/lib/python3.10/site-packages/pydantic/main.py", line 164, in __init__
    __pydantic_self__.__pydantic_validator__.validate_python(data, self_instance=__pydantic_self__)
  File "/code/newape/.venv/lib/python3.10/site-packages/ape/managers/config.py", line 134, in load_configs
    return cm.load()
  File "/code/newape/.venv/lib/python3.10/site-packages/ape/managers/config.py", line 277, in load
    _ = self._plugin_configs
  File "/code/newape/.venv/lib/python3.10/site-packages/ape/managers/config.py", line 205, in _plugin_configs
    valid_ecosystems = dict(self.plugin_manager.ecosystems)
  File "/code/newape/.venv/lib/python3.10/site-packages/ape/plugins/__init__.py", line 138, in __getattr__
    self._register_plugins()
  File "/code/newape/.venv/lib/python3.10/site-packages/ape/plugins/__init__.py", line 193, in _register_plugins
    module = importlib.import_module(module_name)
  File "/usr/lib64/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/code/newape/.venv/lib/python3.10/site-packages/ape_test/__init__.py", line 10, in <module>
    from ape_test.provider import EthTesterProviderConfig, LocalProvider
  File "/code/newape/.venv/lib/python3.10/site-packages/ape_test/provider.py", line 7, in <module>
    from eth.exceptions import HeaderNotFound
  File "/code/newape/.venv/lib/python3.10/site-packages/eth/__init__.py", line 1, in <module>
    import pkg_resources
ModuleNotFoundError: No module named 'pkg_resources'
```

not sure what the TODO should look like, please advise (or code suggest 😉)

Reply:
reset dependencies to how they were in f04d70289737d69453a356398544c917cb547be5